pilot/pkg/ambient/controller/workloadcache.go

```diff
@@ -54,7 +56,19 @@ func initWorkloadCache(opts Options) *workloadCache {
 		_, hasType := o.GetLabels()[ambient.LabelType]
 		return hasType
 	})
-
+	if features.IncludeLocalWorkload {
+		wc.indexes[ambient.TypeWorkload].Insert(ambient.Workload{
+			UID:              "",
+			Name:             "local",
+			Namespace:        "default",
```

Comment:
Ha, wasn't sure if I should include this or not... I ended up putting it behind a feature flag but can remove entirely if we want. The intent is it makes it really easy to run ztunnel on local machine:

* Setup redirect rule to 15001 (I do it on a special uid)
* Run whatever app you want (say `fortio server`)
* Then `sudo -u redirected curl localhost:8080` goes through the ztunnel, does HBONE to itself

Basically we register `localhost` as a workload so we know to send HBONE to it

Reply:
Sounds good, but not in istiod ? As I mentioned in the chat, I think ztunnel should support loading workoad configs from a file - or from a localhost Istiod. 

Any reason to stop using pilot-agent ? We still probably  want DNS, integration with other CA providers ( assuming they 
support same checks as Istiod to allow getting certs for pods on same node ) and all the other stuff.  
Would also simplify the code in ztunnel - and istiod too since we can isolate ztunnel dev/testing from Istiod,